kratos/processes/find_conditions_neighbours_process.cpp

```diff
@@ -0,0 +1,196 @@
+        auto& r_geom = it_cond->GetGeometry();
+        for(unsigned int i = 0; i < r_geom.size(); i++) {
+            (r_geom[i].GetValue(NEIGHBOUR_CONDITIONS)).push_back( Condition::WeakPointer(*(it_cond.base())));
+        }
+    }
+
+    // Shrink to fit
+    block_for_each(r_nodes_array, [](Node& rNode) {
+        rNode.GetValue(NEIGHBOUR_CONDITIONS).shrink_to_fit();
+    });
+
+    // Adding the neighbouring conditions to the condition loop over faces
+    if (mDim == 3) {
+        for(auto it_cond = r_conditions_array.begin(); it_cond!=r_conditions_array.end(); it_cond++) {
+            // Face nodes
+            auto& r_geom = it_cond->GetGeometry();
+            if (r_geom.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D3) { // 3D Triangle
+                // Vector of the 3 faces around the given face
+                auto& r_neighb_faces = it_cond->GetValue(NEIGHBOUR_CONDITIONS);
+                // r_neighb_faces is the vector containing pointers to the three faces around it_cond
+                // r_neighb_faces[0] = neighbour face over edge 1-2 of element it_cond
+                // r_neighb_faces[1] = neighbour face over edge 2-0 of element it_cond
+                // r_neighb_faces[2] = neighbour face over edge 0-1 of element it_cond
+                r_neighb_faces(0) = CheckForNeighbourFaces(r_geom[1].Id(), r_geom[2].Id(), r_geom[1].GetValue(NEIGHBOUR_CONDITIONS), it_cond->Id());
+                r_neighb_faces(1) = CheckForNeighbourFaces(r_geom[2].Id(), r_geom[0].Id(), r_geom[2].GetValue(NEIGHBOUR_CONDITIONS), it_cond->Id());
+                r_neighb_faces(2) = CheckForNeighbourFaces(r_geom[0].Id(), r_geom[1].Id(), r_geom[0].GetValue(NEIGHBOUR_CONDITIONS), it_cond->Id());
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindConditionsNeighboursProcess::Clear()
+{
+    block_for_each(mrModelPart.Nodes(), [](Node& rNode) {
+        if (rNode.Has(NEIGHBOUR_CONDITIONS)) {
+            auto& r_neighbour_conditions = rNode.GetValue(NEIGHBOUR_CONDITIONS);
+            r_neighbour_conditions.erase(r_neighbour_conditions.begin(),r_neighbour_conditions.end());
+        }
+    });
+    block_for_each(mrModelPart.Conditions(), [](Condition& rCond) {
+        if (rCond.Has(NEIGHBOUR_CONDITIONS)) {
+            auto& r_neighbour_conditions = rCond.GetValue(NEIGHBOUR_CONDITIONS);
+            r_neighbour_conditions.erase(r_neighbour_conditions.begin(),r_neighbour_conditions.end());
+        }
+    });
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindConditionsNeighboursProcess::ClearNeighbours()
+{
+    // Call Clear method
```

Comment:
Maybe we can throw a warning in here to inform the user that the `Clear` must be used instead.

Reply:
Done